setup

```diff
@@ -217,24 +255,47 @@ def build(args):
         dst_file = os.path.join(MAGE_BUILD_DIRECTORY, module)
         shutil.copy2(src_file, dst_file)
 
+    logger.info(f"[Terminal] (8/8) Successfully built and copied Rust modules to {MAGE_BUILD_DIRECTORY}.")
+    return True
 
-def run_build_action(args):
-    log.log(level=log.INFO, msg="Starting building of source code...")
-    build(args)
-    log.log(level=log.INFO, msg="Building done.")
+def build(args:Dict[str, Any])->bool:
+    shutil.rmtree(MAGE_BUILD_DIRECTORY, ignore_errors=True)
+    os.makedirs(MAGE_BUILD_DIRECTORY, exist_ok=True)
+    copytree(LICENCE_DIRECTORY, MAGE_BUILD_DIRECTORY)
 
-    if args.path is None:
-        return
+    
+    cpp_status = build_and_copy_cpp_modules(args)
+    if not cpp_status:
+        return False
 
-    log.log(
-        level=log.INFO,
-        msg="Copying build files from folder {src} to {dst} folder.".format(
-            src=MAGE_BUILD_DIRECTORY, dst=args.path
-        ),
-    )
+    python_status = copy_python_modules()
+    if not python_status:
+        return False
+    
+    rust_status = build_copy_rust_modules(args)
```

Comment:
Rename this function to `build_and_copy_rust_modules` in line with `build_and_copy_cpp_modules`